### PR TITLE
fix: Nx.Random + Torchx

### PR DESCRIPTION
**Pull request by polvalente**

### CODE REVIEW

File: nx/lib/nx/random.ex [hunk 1]
```diff
@@ -324,7 +327,7 @@ defmodule Nx.Random do
     |> Nx.reshape(shape, take_names(opts))
   end
 
-  deftransformp randint_random_bits_shape(shape), do: Tuple.insert_at(shape, 0, 2)
+  deftransformp(randint_random_bits_shape(shape), do: Tuple.insert_at(shape, 0, 2))
```
Comment (polvalente) [inline nx/lib/nx/random.ex:330]:
```suggestion
  deftransformp randint_random_bits_shape(shape), do: Tuple.insert_at(shape, 0, 2)
```

File: nx/lib/nx/random.ex [hunk 2]
```diff
@@ -501,7 +514,7 @@ defmodule Nx.Random do
     end
   end
 
-  deftransformp take_names(opts), do: Keyword.take(opts, [:names])
+  deftransformp(take_names(opts), do: Keyword.take(opts, [:names]))
```
Comment (polvalente) [inline nx/lib/nx/random.ex:517]:
```suggestion
  deftransformp take_names(opts), do: Keyword.take(opts, [:names])
```

File: nx/lib/nx/random.ex [hunk 3]
```diff
@@ -519,8 +532,8 @@ defmodule Nx.Random do
     end
   end
 
-  deftransformp put_type(opts, type), do: Keyword.put(opts, :type, type)
-  deftransformp put_shape(opts, shape), do: Keyword.put(opts, :shape, shape)
+  deftransformp(put_type(opts, type), do: Keyword.put(opts, :type, type))
```
Comment (polvalente) [inline nx/lib/nx/random.ex:535]:
```suggestion
  deftransformp put_type(opts, type), do: Keyword.put(opts, :type, type)
```

File: nx/lib/nx/random.ex [hunk 4]
```diff
@@ -519,8 +532,8 @@ defmodule Nx.Random do
...
+  deftransformp(put_shape(opts, shape), do: Keyword.put(opts, :shape, shape))
```
Comment (polvalente) [inline nx/lib/nx/random.ex:536]:
```suggestion
  deftransformp put_shape(opts, shape), do: Keyword.put(opts, :shape, shape)
```

File: nx/lib/nx/random.ex [hunk 5]
```diff
@@ -405,6 +416,8 @@ defmodule Nx.Random do
     normal(key, 0.0, 1.0, opts)
   end
 
+  defnp mask32(tensor), do: Nx.bitwise_and(tensor, 0xFFFFFFFF)
```
Comment (jonatanklosko) [inline nx/lib/nx/random.ex:419]: Perhaps worth a comment why we mask?

File: nx/lib/nx/random.ex [hunk 6]
```diff
@@ -178,14 +181,21 @@ defmodule Nx.Random do
     key1 = ks[0]
     key2 = ks[1]
 
-    xs1 = xs[0] + key1
-    xs2 = xs[1] + key2
+    # slightly different representations (such as representing u32 as s64),
```
Comment (jonatanklosko) [inline nx/lib/nx/random.ex:184]: Having the comment just for `mask32` should be fine, otherwise it can easily get out of sync :)

**Pull request closed without merging**